File: app/src/main/java/com/scp/leagueofquiz/entrypoint/shared/IncrementableLiveData.java
package com.scp.leagueofquiz.entrypoint.shared;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import java.util.function.Consumer;

public class IncrementableLiveData extends DefaultedLiveData<Integer> {

  public IncrementableLiveData(int i) {
    super(i);
  }

  public void setIncrement() {
    increment(this::setValue);
  }

  public void postIncrement() {
    increment(this::postValue);
  }

  public void increment(Consumer<Integer> newValue) {
    newValue.accept(getValue() + 1);
  }
}

File: app/src/main/java/com/scp/leagueofquiz/entrypoint/shared/IncrementableLiveData.kt
package com.scp.leagueofquiz.entrypoint.shared

import java.util.function.Consumer

class IncrementableLiveData(i: Int) : DefaultedLiveData<Int?>(i) {
    fun setIncrement() = increment { this.value = it }

    fun postIncrement() = increment { postValue(it) }

    fun increment(incrementAction: Consumer<Int?>) = incrementAction.accept(value?.plus(1))
}